[member-ordering] Properties with literal keys are wrongly reported in alphabetical order

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

```JSON
{
  "rules": {
    "@typescript-eslint/member-ordering": [2, {
      "default": { "order": "alphabetically" }
    }]
  }
}
```

```TS
interface TestMemberOrder {
  title: string;
  'xyz.foo': string;
}
```

**Expected Result**

The code is valid.

**Actual Result**

The code throws an error due to the literal key of the last property:

```
  3:3  error  Member "xyz.foo" should be declared before member title  @typescript-eslint/member-ordering
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` + `canary` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `14.18.1` |

--- a/packages/eslint-plugin/src/rules/member-ordering.ts
+++ b/packages/eslint-plugin/src/rules/member-ordering.ts
@@ -271,6 +271,31 @@ function getNodeType(node: Member): string | null {
 }
 
 /**
+ * Gets the raw string value of a member's name
+ */
+function getMemberRawName(
+  member:
+    | TSESTree.MethodDefinition
+    | TSESTree.TSMethodSignature
+    | TSESTree.TSAbstractMethodDefinition
+    | TSESTree.PropertyDefinition
+    | TSESTree.TSAbstractPropertyDefinition
+    | TSESTree.Property
+    | TSESTree.TSPropertySignature,
+  sourceCode: TSESLint.SourceCode,
+): string {
+  const { name, type } = util.getNameFromMember(member, sourceCode);
+
+  if (type === util.MemberNameType.Quoted) {
+    return name.substr(1, name.length - 2);
+  }
+  if (type === util.MemberNameType.Private) {
+    return name.substr(1);
+  }
+  return name;
+}
+
+/**
  * Gets the member name based on the member type.
  *
  * @param node the node to be evaluated.
@@ -285,12 +310,12 @@ function getMemberName(
     case AST_NODE_TYPES.TSMethodSignature:
     case AST_NODE_TYPES.TSAbstractPropertyDefinition:
     case AST_NODE_TYPES.PropertyDefinition:
-      return util.getNameFromMember(node, sourceCode).name;
+      return getMemberRawName(node, sourceCode);
     case AST_NODE_TYPES.TSAbstractMethodDefinition:
     case AST_NODE_TYPES.MethodDefinition:
       return node.kind === 'constructor'
         ? 'constructor'
-        : util.getNameFromMember(node, sourceCode).name;
+        : getMemberRawName(node, sourceCode);
     case AST_NODE_TYPES.TSConstructSignatureDeclaration:
       return 'new';
     case AST_NODE_TYPES.TSCallSignatureDeclaration:

--- a/packages/eslint-plugin/tests/rules/member-ordering-alphabetically-order.test.ts
+++ b/packages/eslint-plugin/tests/rules/member-ordering-alphabetically-order.test.ts
@@ -50,6 +50,18 @@ interface Foo {
       options: [{ default: { memberTypes: 'never', order: 'alphabetically' } }],
     },
 
+    // default option + interface + literal properties
+    {
+      code: `
+interface Foo {
+  a : Foo;
+  'b.c' : Foo;
+  "b.d" : Foo;
+}
+            `,
+      options: [{ default: { order: 'alphabetically' } }],
+    },
+
     // default option + type literal + multiple types
     {
       code: `
@@ -84,6 +96,18 @@ type Foo = {
 }
             `,
       options: [{ default: { memberTypes: 'never', order: 'alphabetically' } }],
+    },
+
+    // default option + type + literal properties
+    {
+      code: `
+type Foo = {
+  a : Foo;
+  'b.c' : Foo;
+  "b.d" : Foo;
+}
+            `,
+      options: [{ default: { order: 'alphabetically' } }],
     },
 
     // default option + class + multiple types
@@ -221,6 +245,34 @@ interface Foo {
       ],
     },
 
+    // default option + interface + literal properties
+    {
+      code: `
+interface Foo {
+  "b.d" : Foo;
+  'b.c' : Foo;
+  a : Foo;
+}
+            `,
+      options: [{ default: { order: 'alphabetically' } }],
+      errors: [
+        {
+          messageId: 'incorrectOrder',
+          data: {
+            member: 'b.c',
+            beforeMember: 'b.d',
+          },
+        },
+        {
+          messageId: 'incorrectOrder',
+          data: {
+            member: 'a',
+            beforeMember: 'b.c',
+          },
+        },
+      ],
+    },
+
     // default option + interface + wrong order (multiple)
     {
       code: `
@@ -274,6 +326,34 @@ type Foo = {
           data: {
             member: 'call',
             beforeMember: 'new',
+          },
+        },
+      ],
+    },
+
+    // default option + type + literal properties
+    {
+      code: `
+type Foo = {
+  "b.d" : Foo;
+  'b.c' : Foo;
+  a : Foo;
+}
+            `,
+      options: [{ default: { order: 'alphabetically' } }],
+      errors: [
+        {
+          messageId: 'incorrectOrder',
+          data: {
+            member: 'b.c',
+            beforeMember: 'b.d',
+          },
+        },
+        {
+          messageId: 'incorrectOrder',
+          data: {
+            member: 'a',
+            beforeMember: 'b.c',
           },
         },
       ],

